### ISSUE TITLE
Wrong Output Size for Classification Problems

### ISSUE THREAD

**Issue by orhunguley** [CBICA/GaNDLF#104]
**Describe the bug**
When doing multi-class classification in the current master branch, and the CSV file have the class labels in single column, the number of classes are seen as 1.

**To Reproduce**
1. Checkout to master branch
2. Use the config file attached and prepare the csv file based on the images I attached.
3. Run gandlf

**Expected behavior**
![image](https://user-images.githubusercontent.com/24361544/123286253-0bd99780-d50e-11eb-9b44-b0ae9a04faf9.png)

I think the problem is due to the populate_header_in_parameters function. There is an if statement there that setting the number of classes to the number of prediction headers, which is one header, consisting of the image labels.
![image](https://user-images.githubusercontent.com/24361544/123286581-50fdc980-d50e-11eb-9a83-738b1dea0944.png)

**Screenshots**
[config_class.txt](https://github.com/CBICA/GaNDLF/files/6710191/config_class.txt)

![301004](https://user-images.githubusercontent.com/24361544/122981969-9ef1c080-d39a-11eb-892f-66981378b260.jpg)
![301005](https://user-images.githubusercontent.com/24361544/122981970-9ef1c080-d39a-11eb-90fe-bb97b1927b08.jpg)
![301006](https://user-images.githubusercontent.com/24361544/122981971-9f8a5700-d39a-11eb-91d8-c3fc17bec1c3.jpg)
![301007](https://user-images.githubusercontent.com/24361544/122981973-9f8a5700-d39a-11eb-92cd-10fc0f8ac06e.jpg)
![301000](https://user-images.githubusercontent.com/24361544/122981976-9f8a5700-d39a-11eb-8f2f-a22bee6c5f8d.jpg)
![301001](https://user-images.githubusercontent.com/24361544/122981978-9f8a5700-d39a-11eb-898d-a3d77e1a8a0e.jpg)
![301002](https://user-images.githubusercontent.com/24361544/122981981-a022ed80-d39a-11eb-9faf-f3496a4c11e6.jpg)
![301003](https://user-images.githubusercontent.com/24361544/122981982-a022ed80-d39a-11eb-9a91-53ef5a8dff3b.jpg)
ages.githubusercontent.com/24361544/122981569-31de2b00-d39
[mini_training_data - Copy.txt](https://github.com/CBICA/GaNDLF/files/6710203/mini_training_data.-.Copy.txt)
a-11eb-9ba0-4872c0b1079f.png)
**GaNDLF Version**
0.0.10

**GaNDLF Version**
0.0.10

**Desktop (please complete the following information):**
 - Ubuntu 20.04 WSL 2

**Comment by sarthakpati** [comment]
Can you try after commenting out this line in your config: `class_list: ["0", "1", "2", "3"]`

**Comment by orhunguley** [comment]
@sarthakpati @sarthakpati  Got the same error after removing. Output shape is still in shape of (batch_size, 1).

**Comment by sarthakpati** [comment]
Hmm, you are right, the `num_classes` variable is getting overridden [[ref](https://github.com/CBICA/GaNDLF/blob/master/GANDLF/parameterParsing.py#L266)]. Can you see what happens if you put the following: `class_list: [1]`

**Comment by sarthakpati** [comment]
Hi @orhunguley, upon looking closer, this does seem to be right to me. The output should be of shape `[batch_size,1]` so that it can be processed correctly (since the input label is in the size `[batch_size,1]`. Am I missing something, here?

**Comment by orhunguley** [comment]
Hi @sarthakpati ,
I think the required shape for the output should be (b_size, num_classes), which will include the probability for each class. (b_size, 1) would be valid for binary classification.
For example, if you check the cross entropy documentation in [Pytorch](https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html), the required parameter size is stated as (N, C) for output and (N) for input. Because we should optimize over the probabilities of negative examples as well.

**Comment by sarthakpati** [comment]
Hmm, that does make sense. Reading a few more resources: 
- https://medium.com/swlh/cross-entropy-loss-in-pytorch-c010faf97bab
- https://pytorch.org/tutorials/beginner/blitz/cifar10_tutorial.html
- https://stackoverflow.com/questions/58063826/compute-cross-entropy-loss-for-classification-in-pytorch
- https://towardsdatascience.com/pytorch-tabular-multiclass-classification-9f8211a123ab

Yes, indeed the output should be `(batch_size, num_classes)`. Can you check by making this change manually if it works? If that is the case, then we can incorporate it in a cleaner way into the code.

**Comment by orhunguley** [comment]
Hi @sarthakpati ,

Yes, I already hardcoded and make it work, also implemented a Cross Entropy loss, which is working in training phase. But in validation phase, I am still getting error and working on it right now.
